osquery/filesystem/windows/fileops.cpp

```diff
@@ -98,6 +98,43 @@ class WindowsFindFiles {
   fs::path path_;
 };
 
+Status windowsShortPathToLongPath(const std::string& shortPath,
+                                  std::string& rLongPath) {
+  TCHAR longPath[MAX_PATH];
+  auto ret = GetLongPathName(shortPath.c_str(), longPath, MAX_PATH);
+  if (ret == 0) {
+    return Status(GetLastError(), "Failed to convert short path to long path");
+  }
+  rLongPath = std::string(longPath);
+  return Status();
+}
+
+Status windowsGetFileVersion(const std::string& path, std::string& rVersion) {
+  DWORD handle = 0;
+  auto verSize = GetFileVersionInfoSize(path.c_str(), &handle);
+  auto verInfo = std::make_unique<BYTE[]>(verSize);
+  if (verInfo == nullptr) {
+    return Status(1, "Failed to malloc for version info");
+  }
+  auto err = GetFileVersionInfo(path.c_str(), handle, verSize, verInfo.get());
+  if (err == 0) {
+    return Status(GetLastError(), "Failed to get file version info");
+  }
+  VS_FIXEDFILEINFO* pFileInfo = nullptr;
+  UINT verInfoSize = 0;
+  err = VerQueryValue(
+      verInfo.get(), TEXT("\\"), (LPVOID*)&pFileInfo, &verInfoSize);
```

Comment:
The spec does not indicate this is an optional param (https://msdn.microsoft.com/en-us/library/windows/desktop/ms647464(v=vs.85).aspx), although when I tested it with a null pointer it still worked. But I'd prefer to not make the assumption :).